rust/src/builtins/compose/commit.rs

```diff
@@ -0,0 +1,36 @@
+//! CLI sub-command `compose commit`.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use crate::cxxrsutil::CxxResult;
+use anyhow::anyhow;
+use fn_error_context::context;
+
+#[context("Writing commit-id to {}", target_path)]
+pub fn write_commit_id(target_path: &str, revision: &str) -> CxxResult<()> {
+    if target_path.is_empty() {
+        return Err(anyhow!("empty target path").into());
+    }
+    if revision.is_empty() {
+        return Err(anyhow!("empty revision content").into());
```

Comment:
Fine as is, I probably would have made this a `panic!` instead because I don't believe we could reach this through user error, only an internal logic bug.